src/Microsoft.AspNetCore.ApplicationInsights.HostingStartup/JavaScriptSnippetTagHelperComponent.cs

```diff
@@ -0,0 +1,46 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Threading.Tasks;
+using Microsoft.ApplicationInsights.AspNetCore;
+using Microsoft.AspNetCore.Razor.TagHelpers;
+
+namespace Microsoft.AspNetCore.ApplicationInsights.HostingStartup
+{
+    /// <summary>
+    /// The <see cref="TagHelperComponent"/> that injects the <see cref="JavaScriptSnippet"/> to the head tag.
+    /// </summary>
+    public class JavaScriptSnippetTagHelperComponent : TagHelperComponent
```

Comment:
I've been known for suffixing all my types/names with base types as well but this one in particular feels overly long without conveying what it actually does.

Maybe it doesn't matter though, thinking about it a bit more, is there any reason in it being public? If not, we can mark it internal and the name doesn't matter as much.